src/core.ts

```diff
@@ -93,26 +92,32 @@ export function addEvent(type: string, eventState: any, time?: number) {
   nextPayloadLength += eventStr.length;
 
   // Edge case:
-  // Don't reschedule upload when next payload consists of exactly one XhrError instrumentation event.
+  // Don't schedule next upload when next payload consists of exactly one XhrError instrumentation event.
   // This helps us avoid the infinite loop in the case when all requests fail (e.g. dropped internet connection)
   // Infinite loop comes from sending instrumentation about failing to deliver previous delivery failure instrumentation.
-  let rescheduleUpload = !(eventState && eventState.type === Instrumentation.XhrError && nextPayload.length === 1);
-  if (rescheduleUpload) {
-    if (timeout) {
-      clearTimeout(timeout);
-    }
+  scheduleUpload = scheduleUpload && !(event.state && event.state.type === Instrumentation.XhrError && nextPayload.length === 1);
+  if (scheduleUpload) {
+    clearTimeout(timeout);
     timeout = setTimeout(uploadNextPayload, config.delay);
   }
 }
 
+export function addMultipleEvents(events: IEventData[]) {
+  for (let i = 0; i < events.length; i++) {
+    // Only schedule upload after the last event
+    let scheduleUpload = (i === events.length - 1);
+    addEvent(events[i], scheduleUpload);
```

Comment:
probably a really minor optimization, but we always know we want scheduleUpload to be true on the final iteration so we should just loop until i < events.length - 1 and always pass false within the loop, then make one more explicit call after with events[events.length-1] and true